rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+
+## But who wants this?
+
+A common refrain in response to some of the positive properties that "decentralization" might enable is "none of our existing users asked for this". 
+The truth is that many of these features seem far out of reach in the current architecture of the internet. The way users have interacted with the internet, shaped by the business interests of large tech companies, has told them for three decades these features are not possible. Even so, we've already spoken to potential customers who've expressed concrete interest in this kind of product.
+
+Content addressing provides the core from which to start building toward something better. UCAN is another key component. Filecoin is another part of the puzzle. Blockchains in general are a key component (yes trustless distributed consensus matters). Our job is to fill in the remaining missing pieces.
+
+## Next steps
+
+Taken as a whole, yes, this is absurdly ambitious. It IS a more complex product (especially since it's a network, not a product). Getting every property on the list above is beyond a 1 year scope. Also, W3S IS going to focus on other product features besides "decentralization", particularly those important to market segments we can get to while we build towards our end state.
+
+But, yes, we're going to try to build this. So given that, what missing pieces SHOULD be built in the next six months? 
+
+**This is a loosely prioritized of inituitions. It's not final and it's open to discussions.**
+
+*Btw, we're not alone here -- because we're ultimately building a network, a critical one to the overal PLVerse vision, we can draw from other teams, and eventually, some of the required components may not be built by the W3S*
+
+### Other implementations 
+
+The first step on the path for W3S is indeed to build other implementations of w3up. Local.storage is our first experiment. It would be nice to produce a cloudflare only implementation. The architecture proposed in [Core Node Architecture](https://w3sat.notion.site/Filecoin-SPs-in-the-W3S-Network-16a9b66e7ef3425bbee390c6481bd752?pvs=4) is a kind of aspirational target -- an ultraportable implementation that could run in most server environments, be ultra-sandboxed, provide great update mechanisms, and support rapid iteration.
+
+We should constinue to explore these options.
+
+### Router.storage
+
+We need an implementation of W3up that simply routes to other instances of W3Up. Building this implies several tasks:
+- deciding who gets each piece of content
+- how to manage UCAN delegations across relays
+
+And eventually:
+- how to allow nodes to join and how to decide when to kick them out
+
+But first, we should probably just hardcode a list of nodes.
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like router.storage eventually operate through distributed consensus and produce verifiable results.
```

Comment:
> operate through distributed consensus

In this future, isn't the consensus mechanism itself then a point of centralization and a single point of failure? If the blockchain that does this consensus is down, does that mean the network stops working? A storage network like this feels like something that should be able to splinter and fragment to adapt to the needs of sub-groups, and a single (even permissionless and distributed) consensus mechanism feels to me like it sits in tension with some of the user-facing benefits of "decentralization." Maybe IPC solves this!

Reply:
Personally I do think some of the large L1s can be considered pretty reliable at this point.

Filecoin hasn't halted since 2020
Ethereum went a year without halting (surprising though it had one this recently)
Bitcoin's last halt was apparently long enough ago that I can't find any information on it.

I will assure everyone we will not be building shit on Solana which breaks all the time. :P

Reply:
re: other stuff -- again, I think there's a complicated issue around whether people want choice or simply something that distributes things fairly. I just disagree there isn't value to consensus in this.

Reply:
> disagree there isn't value to consensus in this

totally on the same page - I probably come off as a blockchain hater but inside of me is a computer scientist that thinks they are the coolest thing and desperately wants to find really cool ways to use them. I've just found that it's safer to be a blockchain skeptic by default and to try solving problems nearly any other way before committing to this route. I genuinely hold a lot of hope that we have a real usecase for them here!

Reply:
Here is a the hot take: Job of the router is to uphold user preferences and load balance supply and demand. Assuming that is accurate do we even need blockchain, could that simply be an IPVM task handling routing ? That way users gain a choice to pick the "router" that they find to be a right fit. In fact user with right technical skill could even fork the router to create one that fits their needs if one does not exist already.

Reply:
Yea, I'm open to ideas here. The first task is just to build a router of any kind. If it will run in easily in a WASM environment, bonus, cause then there is a path to either IPC OR IPVM, once the right path becomes clear.

Reply:
So if someone wants to write the router in rust, I'm open to it. As long as it doesn't use rust async, which I think won't work well in either env.